File: javascript/browser/src/samples/linguisticOptions/LinguisticOptionsService.js
import SdkFactory, {
    ClientConfiguration,
    CredentialsConfiguration,
    GetLinguisticOptionsRequest
} from "@language-weaver/lw-sdk-js";

export const getLinguisticOptions = async (clientId, clientSecret) => {
    try {
        const clientConfiguration = new ClientConfiguration();
        clientConfiguration.credentialsConfiguration = new CredentialsConfiguration(clientId, clientSecret);
        const lwClient = await SdkFactory.getLanguageWeaverClient(clientConfiguration);
        const request = new GetLinguisticOptionsRequest();
        request.sourceLanguageId = "chi";
        request.targetLanguageId = "eng";
        request.model = "generic";
        const linguisticOptionsResult = await lwClient.getLinguisticOptions(request);
        console.log(linguisticOptionsResult.linguisticOptions);
    } catch (e) {
        console.log(e);
    }
}

File: java/src/main/java/com/languageweaver/sdk/samples/linguisticoptions/LinguisticOptionsService.java
package com.languageweaver.sdk.samples.linguisticoptions;

import com.languageweaver.sdk.common.LanguageWeaverClient;
import com.languageweaver.sdk.common.SdkFactory;
import com.languageweaver.sdk.common.configurations.ClientConfiguration;
import com.languageweaver.sdk.linguisticoptions.common.request.GetLinguisticOptionsRequest;
import com.languageweaver.sdk.linguisticoptions.common.result.LinguisticOptionsResult;

public class LinguisticOptionsService {

    public static void main(String[] args) throws Exception {
        try (LanguageWeaverClient lwClient = new SdkFactory().getLanguageWeaverClient(new ClientConfiguration())) {
            GetLinguisticOptionsRequest getLinguisticOptionsRequest = new GetLinguisticOptionsRequest()
                    .setSourceLanguageId("chi")
                    .setTargetLanguageId("eng")
                    .setModel("generic");

            final LinguisticOptionsResult linguisticOptionsResult = lwClient.getLinguisticOptions(getLinguisticOptionsRequest);
            // handle result
        }
    }
}
